resharper/src/main/java/org/sonarlint/intellij/resharper/AnalyzerConfiguration.java

```diff
@@ -0,0 +1,312 @@
+    properties.put("builtinHeaders",
+      compilerSettings.getHeadersSearchPaths().stream().filter(HeadersSearchPath::isBuiltInHeaders).map(HeadersSearchPath::getPath).collect(Collectors.joining("\n")));
+  }
+
+  @NotNull
+  private static String getPreprocessorDefines(OCCompilerSettings compilerSettings) {
+    Object result;
+    try {
+      result = getPreprocessorDefinesMethod().invoke(compilerSettings);
+    } catch (ReflectiveOperationException e) {
+      throw new IllegalStateException(e);
+    }
+    if (result instanceof List) {
+      return String.join("\n", (List<String>) result) + "\n";
+    } else if (result instanceof String) {
+      return result + "\n";
+    } else {
+      throw new IllegalStateException(result.toString());
+    }
+  }
+
+  @Nullable
+  static String mapToCFamilyCompiler(OCCompilerKind compilerKind) {
+    return switch (compilerKind.getDisplayName()) {
+      case "AppleClang", "Clang", "GCC" -> "clang";
+      case "clang-cl" -> "clang-cl";
+      case "MSVC" -> "msvc-cl";
+      default -> null;
+    };
+  }
+
+  @Nullable
+  static CLanguageKind getLanguageKind(Language language) {
+    if (language.getDisplayName().equals(CLanguageKind.C.getDisplayName())) {
+      return CLanguageKind.C;
+    } else if (language.equals(CppLanguage.INSTANCE)) {
+      return CLanguageKind.CPP;
+    } else if (language.getDisplayName().equals(CLanguageKind.OBJ_C.getDisplayName())) {
+      return CLanguageKind.OBJ_C;
+    } else {
+      return null;
+    }
+  }
+
+  @Nullable
+  static ForcedLanguage getSonarLanguage(Language language) {
+    if (language.getDisplayName().equals(CLanguageKind.C.getDisplayName())) {
+      return ForcedLanguage.C;
+    } else if (language.equals(CppLanguage.INSTANCE)) {
+      return ForcedLanguage.CPP;
+    } else if (language.getDisplayName().equals(CLanguageKind.OBJ_C.getDisplayName())) {
+      return ForcedLanguage.OBJC;
+    } else {
+      return null;
+    }
+  }
+
+  private boolean usingRemoteOrWslToolchain(OCResolveConfiguration configuration) {
+    final var initializedWorkspaces = CidrWorkspace.getInitializedWorkspaces(project);
+    CPPEnvironment cppEnvironment = null;
+    for (var initializedWorkspace : initializedWorkspaces) {
+      if (initializedWorkspace instanceof CMakeWorkspace cMakeWorkspace) {
+        cppEnvironment = getCMakeCppEnvironment(cMakeWorkspace, configuration);
+      } else {
+        cppEnvironment = tryReflection(initializedWorkspace);
+        if (cppEnvironment != null) {
+          break;
+        }
+      }
+    }
+    return cppEnvironment != null && (cppEnvironment.getToolSet().isRemote() || cppEnvironment.getToolSet().isWSL() || cppEnvironment.getToolSet().isDocker());
+  }
+
+  @Nullable
+  private CPPEnvironment tryReflection(CidrWorkspace initializedWorkspace) {
+    // Use reflection to check if workspace is instanceof com.jetbrains.cidr.project.workspace.WorkspaceWithEnvironment interface has
+    // getEnvironment() method
```

Comment:
Do we still need to use reflection in the `resharper` module? Also we try to get things from `com.jetbrains.cidr` I thought this was not present anymore

Reply:
Not everything is gone from `com.jetbrains.cidr`, it's only specific packages, there is no official documentation about the changes so we discover case by case